recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -390,48 +384,60 @@ def validate(self):
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
+    def generate(self):
+        tc = CMakeToolchain(self)
+        # All option() are defined before project() in upstream CMakeLists,
+        # therefore we must use cache_variables
 
         build_only = ["core"]
         for sdk in self._sdks:
             if self.options.get_safe(sdk):
                 build_only.append(sdk)
-        self._cmake.definitions["BUILD_ONLY"] = ";".join(build_only)
+        tc.cache_variables["BUILD_ONLY"] = ";".join(build_only)
 
-        self._cmake.definitions["ENABLE_UNITY_BUILD"] = True
-        self._cmake.definitions["ENABLE_TESTING"] = False
-        self._cmake.definitions["AUTORUN_UNIT_TESTS"] = False
-        self._cmake.definitions["BUILD_DEPS"] = False
+        tc.cache_variables["ENABLE_UNITY_BUILD"] = True
+        tc.cache_variables["ENABLE_TESTING"] = False
+        tc.cache_variables["AUTORUN_UNIT_TESTS"] = False
+        tc.cache_variables["BUILD_DEPS"] = False
         if self.settings.os != "Windows":
-            self._cmake.definitions["ENABLE_OPENSSL_ENCRYPTION"] = True
-
-        self._cmake.definitions["MINIMIZE_SIZE"] = self.options.min_size
-        if self._is_msvc and not self._use_aws_crt_cpp:
-            self._cmake.definitions["FORCE_SHARED_CRT"] = "MD" in msvc_runtime_flag(self)
-
-        if tools.cross_building(self):
-            self._cmake.definitions["CURL_HAS_H2_EXITCODE"] = "0"
-            self._cmake.definitions["CURL_HAS_H2_EXITCODE__TRYRUN_OUTPUT"] = ""
-            self._cmake.definitions["CURL_HAS_TLS_PROXY_EXITCODE"] = "0"
-            self._cmake.definitions["CURL_HAS_TLS_PROXY_EXITCODE__TRYRUN_OUTPUT"] = ""
-        self._cmake.configure()
-        return self._cmake
+            tc.cache_variables["ENABLE_OPENSSL_ENCRYPTION"] = True
+
+        tc.cache_variables["MINIMIZE_SIZE"] = self.options.min_size
+        if is_msvc(self) and not self._use_aws_crt_cpp:
+            tc.cache_variables["FORCE_SHARED_CRT"] = not is_msvc_static_runtime(self)
+
+        if cross_building(self):
+            tc.cache_variables["CURL_HAS_H2_EXITCODE"] = "0"
+            tc.cache_variables["CURL_HAS_H2_EXITCODE__TRYRUN_OUTPUT"] = ""
+            tc.cache_variables["CURL_HAS_TLS_PROXY_EXITCODE"] = "0"
+            tc.cache_variables["CURL_HAS_TLS_PROXY_EXITCODE__TRYRUN_OUTPUT"] = ""
+        if is_msvc(self):
+            tc.preprocessor_definitions["_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING"] = "1"
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
```

Comment:
Do you need this? I thought the idiomatic way was adding the generator name to the ```generators``` variable?

Reply:
There is no idiomatic way, and I prefer to centralize generators in the same place like we do in all other recipes.